.github/workflows/build.yml

```diff
@@ -0,0 +1,36 @@
+name: Deploy the website
+
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+    branches:
+      - main
+
+permissions:
+  contents: write
+
+jobs:
+  build-and-deploy:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v3
+       
+      - name: Install Nix package manager
+        uses: DeterminateSystems/nix-installer-action@main
+
+      - name: Build the website
+        run: |
+          nix develop --command cargo run --verbose
+          nix develop --command cargo build --verbose
+
+      - name: Deploy build to gh-pages branch
+        uses: crazy-max/ghaction-github-pages@v4
+        with:
+          target_branch: gh-pages
+          build_dir: docs
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
```

Comment:
so I should create a token with a limited scope to use this auto-builder, got it. I'll audit the action code before doing this.

Reply:
Yo do not necessarily have to use GitHub auth token if you allow `read and write` perms?